internal/storage/v2/clickhouse/tracestore/reader.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package tracestore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel"
+)
+
+const (
+	selectAllServices = `SELECT name FROM services`
```

Comment:
I'd prefix constants with "sql".

Does this need DISTINCT?

Reply:
@yurishkuro I've added it for now but we might be able to remove it later if we just add it to the materialized view.